### PR TITLE
build: use backtrace package, add option to include Debug symbols

### PR DESCRIPTION
**Pull request by vishwa2710**
* Switch off Clang format as default until we have made a decision on how to proceed
* Add debug symbols option, ON by default
* Use boost backtrace package for debugging

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -280,17 +288,19 @@ IF (BUILD_SHARED_LIBRARY)
...
 
     SET_TARGET_PROPERTIES ( ${SHARED_LIBRARY_TARGET} PROPERTIES
                             VERSION ${PROJECT_VERSION_STRING}
                             SOVERSION ${PROJECT_VERSION_MAJOR}
                             OUTPUT_NAME ${SHARED_LIBRARY_NAME}
                             CLEAN_DIRECT_OUTPUT 1
+                            ENABLE_EXPORTS ON
                             INSTALL_RPATH "$ORIGIN/../lib:$ORIGIN/")
 
+                            # set_property(TARGET your_target PROPERTY ENABLE_EXPORTS ON
```
Comment (kyle-cochran) [inline CMakeLists.txt:302]:
```suggestion
```

File: test/OpenSpaceToolkit/Core/FileSystem/Directory.test.cpp [hunk 2]
```diff
@@ -117,6 +117,7 @@ TEST(OpenSpaceToolkit_Core_FileSystem_Directory, Exists)
     {
         EXPECT_ANY_THROW(Directory::Undefined().exists());
     }
+
```
Comment (kyle-cochran) [inline test/OpenSpaceToolkit/Core/FileSystem/Directory.test.cpp:120]:
```suggestion
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -278,10 +295,8 @@ IF (BUILD_SHARED_LIBRARY)
     TARGET_INCLUDE_DIRECTORIES (${SHARED_LIBRARY_TARGET} PUBLIC "${PROJECT_SOURCE_DIR}/include/")
     TARGET_INCLUDE_DIRECTORIES (${SHARED_LIBRARY_TARGET} PUBLIC "${PROJECT_SOURCE_DIR}/src/")
 
-    TARGET_LINK_LIBRARIES (${SHARED_LIBRARY_TARGET} "stdc++")
     TARGET_LINK_LIBRARIES (${SHARED_LIBRARY_TARGET} "pthread")
-    TARGET_LINK_LIBRARIES (${SHARED_LIBRARY_TARGET} "Boost::filesystem" "Boost::log" ${Boost_STACKTRACE_BASIC_LIBRARY})
-    TARGET_LINK_LIBRARIES (${SHARED_LIBRARY_TARGET} "dl")
```
Comment (kyle-cochran) [inline CMakeLists.txt:284]: This was needed for stacktrace basic, surprised it's not needed for backtrace.

Comment (kyle-cochran) [reply CMakeLists.txt:284]: Confirmed it works even with a fresh docker environment 👍. All good

**Review by kyle-cochran — APPROVED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (vishwa2710) [reply CMakeLists.txt:284]: Awesome thanks for checking.

**Pull request merged**